ALPHA_BETA.md

```diff
@@ -0,0 +1,109 @@
+# rubot: alpha_beta
+
+The goal of the `Bot` is to take a decision tree and output
+the optimal decision.
+
+At each depth of this tree we either want to maximise or minimize
+the resulting fitness.
+
+In a competitive two player game, maximizing depths are the turns in which the bot
+is active, and during minimizing depths the opponent is able to make a decision.
+
+- `a n` := the bot is active, meaning we want to take the path which results in the highest fitness.
+    the node itself has a fitness of `n`.
+- `o n` := the opponent is active, meaning that we want to achieve the lowest possible fitness.
+
+Paths are represented as a list of indices.
+
+## Examples
+
+```
+a 0
+  - a 12 # child 0
+  - a 7  # child 1
+
+optimal action: [0], with fitness: 12
+
+a 0
+  - a 12
+    - a 6
+  - a 7
+
+optimal action: [1], with fitness: 7
+
+a 0
+  - o 12
+    - a 3
+    - a 7
+  - a 6
+
+optimal action: [1], with fitness: 6
```

Comment:
Here we calculate minimum of children fitnesses for `o` node and then maximum for `a` root node, makes sense.